parse.go

```diff
@@ -120,12 +121,16 @@ func NewParser(in io.Reader, bytesToRead int64, frameChannel chan *frame.Frame)
 		frameChannel: frameChannel,
 	}
 
-	debug.Log("NewParser: readHeader")
-	elems, err := p.readHeader()
-	if err != nil {
-		return nil, err
+	elems := []*Element{}
+
+	if !optSet.skipMetadataReadOnNewParserInit {
```

Comment:
Oh hey actually one question. I didn't recall this earlier, but we do have some "fallback" code that automatically skips header reading if the DICOM magic word isn't found at position 128: https://github.com/suyashkumar/dicom/blob/a3cee26ddc019e03eceb847fbc6eb54aa4256db3/parse.go#L196-L203

Would this fallback behavior automatically work for your use case without explicitly specifying the skipMetadataReadOnNewParserInit (or needing the option)?  (Even if so, we could probably document this better). 

Apologies for not thinking of this earlier.

Reply:
For most scenarios I think this will work. For very short headerless datasets however, `p.reader.Peek(128 + 4)` will return an error, stopping NewParser at line 130. For what I'm doing I will sometimes handle datasets containing only one or two elements, so I think having the `skipMetadataReadOnNewParserInit` option is valuable.

Reply:
Gotcha, thanks for the info. This seems reasonable, we can leave it as is.